### PR TITLE
Feature/ra 81

### PR DESCRIPTION
**Pull request by keaaa**

### CODE REVIEW

File: api/applications/applications_handler.go [hunk 1]
```diff
@@ -27,26 +27,32 @@ import (
...
 	return ApplicationHandler{
-		userAccount:        accounts.UserAccount,
-		serviceAccount:     accounts.ServiceAccount,
+		accounts:           accounts,
 		jobHandler:         jobHandler,
 		environmentHandler: environments.Init(accounts),
 	}
 }
 
+func (ah ApplicationHandler) userAccount() models.Account {
```
Comment (thezultimate) [inline api/applications/applications_handler.go]: `userAccount()` -> `getUserAccount()` ?

File: api/applications/applications_handler.go [hunk 2]
```diff
@@ -27,26 +27,32 @@ import (
...
+	return ah.accounts.UserAccount
+}
+
+func (ah ApplicationHandler) serviceAccount() models.Account {
```
Comment (thezultimate) [inline api/applications/applications_handler.go]: `serviceAccount()` -> `getServiceAccount()` ?

File: api/environments/environment_controller_test.go [hunk 3]
```diff
@@ -1063,7 +1063,7 @@ func TestStopStartRestartComponent_ApplicationWithDeployment_EnvironmentConsiste
 }
 
 func initHandler(client kubernetes.Interface, radixclient radixclient.Interface) EnvironmentHandler {
-	return Init(models.NewAccounts(client, radixclient, client, radixclient))
+	return Init(models.NewAccounts(client, radixclient, client, radixclient, "", models.NullObjImpersonation()))
```
Comment (thezultimate) [inline api/environments/environment_controller_test.go]: `models.NullObjImpersonation()` -> `models.Impersonation{}` ?

File: api/jobs/job_controller_test.go [hunk 4]
```diff
@@ -90,7 +90,7 @@ func TestGetApplicationJob(t *testing.T) {
 		PushImage: true,
 	}
 
-	handler := Init(models.NewAccounts(client, radixclient, client, radixclient))
+	handler := Init(models.NewAccounts(client, radixclient, client, radixclient, "", models.NullObjImpersonation()))
```
Comment (thezultimate) [inline api/jobs/job_controller_test.go]: `models.NullObjImpersonation()` -> `models.Impersonation{}` ?

File: api/utils/kubernetes.go [hunk 5]
```diff
@@ -48,27 +47,23 @@ func NewKubeUtil(useOutClusterClient bool) KubeUtil {
 // GetOutClusterKubernetesClient Gets a kubernetes client using the bearer token from the radix api client
 func (ku *kubeUtil) GetOutClusterKubernetesClient(token string) (kubernetes.Interface, radixclient.Interface) {
 	if ku.useOutClusterClient {
-		config, _ := getOutClusterClientConfig(token, nil, nil)
+		config := getOutClusterClientConfig(token, models.NullObjImpersonation())
```
Comment (thezultimate) [inline api/utils/kubernetes.go]: `models.NullObjImpersonation()` -> `models.Impersonation{}` ?

File: models/impersonation.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+func NewImpersonation(user, group string) (Impersonation, error) {
+	impersonation := Impersonation{
+		User:  strings.TrimSpace(user),
+		Group: strings.TrimSpace(group),
+	}
+	return impersonation, impersonation.isValid()
+}
+
+func NullObjImpersonation() Impersonation {
```
Comment (thezultimate) [inline models/impersonation.go]: Probably this function is not needed.

File: api/applications/applications_handler.go (hunk 1 above)
Comment (keaaa) [reply api/applications/applications_handler.go]: agree - fixed

File: api/applications/applications_handler.go (hunk 2 above)
Comment (keaaa) [reply api/applications/applications_handler.go]: agree -fixed

**Pull request merged**